exonum/src/node/requests.rs

```diff
@@ -71,26 +74,40 @@ impl NodeHandler {
 
     /// Handles `TransactionsRequest` message. For details see the message documentation.
     pub fn handle_request_txs(&mut self, msg: &Signed<TransactionsRequest>) {
-        use std::mem;
         trace!("HANDLE TRANSACTIONS REQUEST");
+        self.send_transactions_by_hash(&msg.author(), msg.txs());
+    }
+
+    /// Handles `PoolTransactionsRequest` message. For details see the message documentation.
+    pub fn handle_pool_request_txs(&mut self, msg: &Signed<PoolTransactionsRequest>) {
+        trace!("HANDLE POOL TRANSACTIONS REQUEST");
+        let snapshot = self.blockchain.snapshot();
+        let schema = Schema::new(&snapshot);
+        let hashes: Vec<Hash> = schema.transactions_pool().iter().collect();
+
+        self.send_transactions_by_hash(&msg.author(), &hashes);
+    }
+
+    fn send_transactions_by_hash(&mut self, author: &PublicKey, hashes: &[Hash]) {
+        use std::mem;
         let snapshot = self.blockchain.snapshot();
         let schema = Schema::new(&snapshot);
         let mut txs = Vec::new();
         let mut txs_size = 0;
         let unoccupied_message_size = self.state.config().consensus.max_message_len as usize
             - TRANSACTION_RESPONSE_EMPTY_SIZE;
 
-        for hash in msg.txs() {
+        for hash in hashes {
             let tx = get_tx(&hash, &schema.transactions(), &self.state.tx_cache());
             if let Some(tx) = tx {
```

Comment:
I'd write it in one line:
```suggestion
            if let Some(tx) = get_tx(&hash, &schema.transactions(), &self.state.tx_cache()) {
```